yarn-project/circuits.js/src/contract/contract_class_id.ts

```diff
@@ -56,8 +57,19 @@ export type ContractClassIdPreimage = {
   publicBytecodeCommitment: Fr;
 };
 
-// TODO(#5860): Replace with actual implementation
-// Changed to work with canonical contracts that may have non-deterministic noir compiles and we want to keep the address constant
-export function computePublicBytecodeCommitment(_bytecode: Buffer) {
-  return new Fr(5);
+export function computePublicBytecodeCommitment(packedBytecode: Buffer) {
+  // Encode the buffer into field elements (chunked into 31 bytes each)
+  const encodedBytecode: Fr[] = bufferAsFields(packedBytecode, MAX_PACKED_PUBLIC_BYTECODE_SIZE_IN_FIELDS);
+  // The first element is the length of the bytecode (in bytes)
+  const bytecodeLength = Math.ceil(encodedBytecode[0].toNumber() / (Fr.SIZE_IN_BYTES - 1));
+  // Hash it
+  let bytecodeCommitment = new Fr(0);
+  // If the bytecode is longer than what is supported, we'll have problems accessing the encodedBytecode array
```

Comment:
Should be ok to assert that the length is ok right now right?

Reply:
I dont think i want to fail here though, depends on what our intended behavior is.

Someone can request to deploy a contract that is bigger than `MAX_PACKED_PUBLIC_BYTECODE_SIZE_IN_FIELDS` - it just gets truncated. But we should make sure we don't hash all of it - just up to the limit so that we dont fail the circuit proving.

Of course i think this contract would fail if you tried to access a PC > than the max limit. But if a truncated form of the contract is valid - should we allow it to be deployed?